go/integration_test/processor/calls_test.go

```diff
@@ -0,0 +1,91 @@
+package processor
+
+import (
+	"fmt"
+	"math/big"
+	"strings"
+	"testing"
+
+	"github.com/Fantom-foundation/Tosca/go/tosca"
+	"github.com/Fantom-foundation/Tosca/go/tosca/vm"
+)
+
+func TestProcessor_MaximalCallDepthIsEnforced(t *testing.T) {
+	gasLimit := tosca.Gas(1000000000000)
+	for processorName, processor := range getProcessors() {
+		// todo implement recurisve calls and depth checks in floria
+		if strings.Contains(processorName, "floria") {
+			continue
+		}
+
+		t.Run(fmt.Sprintf("%s-MaxCallDepth", processorName), func(t *testing.T) {
+			sender := tosca.Address{1}
+			receiver := &tosca.Address{2}
+
+			values := []*big.Int{big.NewInt(32), big.NewInt(0),
+				big.NewInt(32), big.NewInt(0), big.NewInt(0),
+				new(big.Int).SetBytes(receiver[:]), big.NewInt(int64(gasLimit))}
+			callStack := pushToStack(values)
```

Comment:
nit: the variable name `callStack` is a bit misleading ... since in the end it is code establishing the stack listed above.
Also, it is only used at one position below. Creating it there with something like
```
code = append(code, pushToStack([]*big.Int{
    big.NewInt(32),                                // result size
    big.NewInt(0),                                  // result offset
    big.NewInt(32),                                // argument size
    big.NewInt(0),                                  // argument offset
    big.NewInt(0),                                  // transfered value
    new(big.Int).SetBytes(receiver[:]),   // call target
    big.NewInt(int64(gasLimit)),             // gas send to nested call
})
```
would clean this up. Since this is the reverse order of parameters as usually used when looking at the CALL operation specification, it might also be a nice feature to invert the parameter order.

Wdyt?

Reply:
Nice suggestion, I changed the argument order and appended it directly to the `code`.